src/barfturn.ts

```diff
@@ -0,0 +1,369 @@
+      adventureMacroAuto(determineDraggableZoneAndEnsureAccess(), Macro.basicCombat());
+      return get("lastVoteMonsterTurn") === totalTurnsPlayed();
+    },
+    spendsTurn: false,
+  },
+  {
+    name: "Digitize Wanderer",
+    available: () => Counter.get("Digitize Monster") <= 0,
+    execute: () => {
+      const isEmbezzler = SourceTerminal.getDigitizeMonster() === embezzler;
+      const start = get("_sourceTerminalDigitizeMonsterCount");
+
+      const shouldGoUnderwater =
+        isEmbezzler &&
+        !get("_envyfishEggUsed") &&
+        myLevel() >= 11 &&
+        (getModifier("Adventure Underwater") ||
+          waterBreathingEquipment.some((item) => have(item))) &&
+        (getModifier("Underwater Familiar") ||
+          familiarWaterBreathingEquipment.some((item) => have(item))) &&
+        (have($effect`Fishy`) || (have($item`fishy pipe`) && !get("_fishyPipeUsed"))) &&
+        !have($item`envyfish egg`) &&
+        canAdventure($location`The Briny Deeps`);
+      const targetLocation = shouldGoUnderwater
+        ? $location`The Briny Deeps`
+        : determineDraggableZoneAndEnsureAccess();
+
+      isEmbezzler ? embezzlerPrep() : freeFightPrep();
+      adventureMacroAuto(targetLocation, Macro.basicCombat());
+      return get("_sourceTerminalDigitizeMonsterCount") !== start;
+    },
+    spendsTurn: () => !SourceTerminal.getDigitizeMonster()?.attributes.includes("FREE"),
+  },
+  {
+    name: "Guaranteed Kramco",
+    available: () => sober() && kramcoGuaranteed(),
+    execute: () => {
+      freeFightPrep(new Requirement([], { forceEquip: $items`Kramco Sausage-o-Matic™` }));
+      adventureMacroAuto(determineDraggableZoneAndEnsureAccess(), Macro.basicCombat());
+      return !kramcoGuaranteed();
+    },
+    spendsTurn: false,
+  },
+  {
+    name: "Void Monster",
+    available: () =>
+      sober() &&
+      have($item`cursed magnifying glass`) &&
+      get("cursedMagnifyingGlassCount") === 13 &&
+      get("_voidFreeFights") < 5,
+    execute: () => {
+      freeFightPrep(new Requirement([], { forceEquip: $items`cursed magnifying glass` }));
+      adventureMacroAuto(determineDraggableZoneAndEnsureAccess(), Macro.basicCombat());
+      return get("cursedMagnifyingGlassCount") === 0;
+    },
+    spendsTurn: false,
+  },
+  {
+    name: "Envyfish Egg",
+    available: () =>
+      have($item`envyfish egg`) && get("envyfishMonster") === embezzler && !get("_envyfishEggUsed"),
+    execute: () => {
+      embezzlerPrep();
+      withMacro(Macro.meatKill(), () => use($item`envyfish egg`), true);
+      return get("_envyfishEggUsed");
+    },
+    spendsTurn: true,
+  },
+  {
+    name: "Spit Acid",
+    available: () =>
+      sober() && have($item`Jurassic Parka`) && !have($effect`Everything Looks Yellow`),
+    execute: () => {
+      const jellyfishing = have($familiar`Space Jellyfish`) && realmAvailable("stench");
+      const familiarChoice = jellyfishing ? $familiar`Space Jellyfish` : freeFightFamiliar(true);
```

Comment:
Is there a reason to intentionally run jellyfish over a general marginal familiar (which could very well be the jellyfish)?

Reply:
We don't currently have a way to pass additional familiars to freeFightFamiliar, but I suppose I should just change that